async-nats/src/service/mod.rs

```diff
@@ -199,6 +217,20 @@ pub struct Service {
 }
 impl Service {
     async fn add(client: Client, config: Config) -> Result<Service, Error> {
```

Comment:
In Rust, if stream is expected to be empty, it is `Option<String>`, so I would consider this a defensive programming.
Though when we wil introduce `Subject` type this will be improved automatically.